proto/dydxprotocol/clob/query.proto

```diff
@@ -220,3 +220,38 @@ message StreamOrderbookFill {
   // Resulting fill amounts for each order in the orders array.
   repeated uint64 fill_amounts = 3;
 }
+
+// StreamTakerOrder provides information on a taker order that was attempted
+// to be matched on the orderbook.
+// It is intended to be used only in full node streaming.
+message StreamTakerOrder {
+  // The taker order that was matched on the orderbook. Can be a
+  // regular order or a liquidation order.
+  oneof taker_order {
+    Order order = 1;
+    StreamLiquidationOrder liquidation_order = 2;
+  }
+
+  // Information on the taker order after it is matched on the book,
+  // either successfully or unsuccessfully.
+  StreamTakerOrderStatus taker_order_status = 3;
+}
+
+// StreamTakerOrderStatus is a representation of a taker order
+// after it is attempted to be matched on the orderbook.
+// It is intended to be used only in full node streaming.
+message StreamTakerOrderStatus {
+  // The state of the taker order after attempting to match it against the orderbook.
+  // Possible enum values can be found here:
+  // https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/clob/types/orderbook.go#L105
+  uint32 order_status = 1;
+
+  // The amount of remaining (non-matched) base quantums of this taker order.
+  uint64 remaining_quantums = 2;
```

Comment:
for my understanding - this is the quantums remaining after optimisitic fills?

Reply:
yes, after the matching loop, remaining quantums